feat(email): support for a custom envelope

ConnyOnny opened the pull request:
The EmailBuilder now has a function to add a preconfigured envelope,
overriding the auto generated one.

fixes #84

amousset commented on src/email/mod.rs:
```diff
@@ -264,6 +262,34 @@ pub struct Envelope {
+        }
+    }
+    /// Adds a receiver
+    pub fn to<S: Into<String>>(mut self, adr: S) -> Self {
```
Could you rename `adr` to `address` for consistency with other methods?

amousset commented on src/email/mod.rs:
```diff
@@ -575,13 +605,61 @@ impl EmailBuilder {
+                                    // if it's an author group, use the first author
+                                    Some(m) => m.address.clone(),
+                                    // for an empty author group (the rarest of the rare cases)
+                                    None => String::new(), // empty envelope sender
```
If we do not implement the non-compliance flag now, I would prefer keeping the error.

ConnyOnny commented:
I don't have time for this until Saturday. So let's not merge it for now.

ConnyOnny commented:
I have hopefully addressed the above issues. If you agree, please merge it. We can still make behavioral detail changes, once we have a good strategy for issue #95
